subprojects/language-groovy/src/main/java/org/gradle/api/tasks/javadoc/Groovydoc.java

```diff
@@ -268,16 +269,42 @@ public void setFooter(String footer) {
 
     /**
      * Returns a HTML file to be used for overview documentation. Set to {@code null} when there is no overview file.
+     *
+     * @deprecated please use {@link #getOverviewText()} instead
      */
+    @Deprecated
```

Comment:
In addition to marking the method as deprecated, you also need to use the SingleMessageLogger.nagUserOfDeprecated() method [like in the sample I mentioned](https://github.com/gradle/gradle/blob/master/subprojects/scala/src/main/groovy/org/gradle/api/tasks/scala/ScalaCompileOptions.java#L62).

This is also true for the setter.